naga/src/back/hlsl/writer.rs

```diff
@@ -1602,6 +1602,16 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
         use crate::Statement;
 
         match *stmt {
+            Statement::Phony(expr) => {
+                write!(self.out, "{level}")?;
+                self.write_named_expr(
+                    module,
+                    expr,
+                    format!("_phony_{}", expr.index()),
+                    expr,
+                    func_ctx,
+                )?;
+            }
```

Comment:
question: @gfx-rs/naga, should we be using `self.namer` here? I think so, since I don't think this logic has any guarantee of a hygienic name?

Reply:
The catch, as hinted at in https://github.com/gfx-rs/wgpu/issues/6095#issuecomment-2363993831, is that even ignored expressions affect WGSL's rules for which declarations are [statically accessed](https://gpuweb.github.io/gpuweb/wgsl/#statically-accessed). In other words, when you've got

```
let _  = buffer;
```

in some function body, according to WGSL's rules, that means that `buffer` is "statically accessed" by that function - even though the expression is ignored.